lib/dal/src/attribute/value.rs

```diff
@@ -2162,8 +2164,14 @@ impl AttributeValue {
         func_id: Option<FuncId>,
     ) -> AttributeValueResult<()> {
         let func_id = if let Some(id) = func_id {
+            let func = Func::get_by_id(ctx, id).await?;
+            if !func.is_transformation && !func.builtin {
+                return Err(AttributeValueError::SubscribingWithInvalidFunction);
+            }
+
             id
         } else {
+            // TODO(victor) remove this new ui comes around
```

Comment:
I think we can remove this now along with the associated functionality in the new UI (not in this PR, just saying I think rolling back "subscribe to an array with a scalar" as a function in the old UI).